### ISSUE TITLE
http_listener::support method and HTTP OPTIONS(or TRACE)

### ISSUE THREAD

**Issue by JengdiB** [Microsoft/cpprestsdk#158]
Hi,

I don't know why the `http_listener::support(const std::function<void(http_request)> &)` method does not forward OPTIONS(or TRACE) requests to a given handler. I have to use another method to clearly specify that I want to support those two verbs.

I noticed that in http_listener_impl class there are 
```
// Default implementation for TRACE and OPTIONS.
void handle_trace(http::http_request message);
void handle_options(http::http_request message);
```

so please enlighten me the reason why Cassablanca have default handlers for them?

**Comment by ras0219-msft** [comment]
We suspect that the vast majority of use cases do not want to individually implement those two request types and would benefit from the common implementation we've given (of course, with the possibility to override).

Could you go into more detail about how this doesn't work for your situation?

**Comment by JengdiB** [comment]
I'm using Cassablanca to implement CORS-enabled REST Server so I have to handle Preflight request from web browser. This request is issued with 'OPTIONS' as HTTP Method that I believe that if I will receive the request by registering the listener to listen to all request by using the `http_listener::support(const std::function<void(http_request)> &)` method. But it turned out that I did not receive any 'OPTIONS' requests. I dig around the http_listener.h and found two default methods to handle TRACE and OPTIONS in http_listener_impl class. So I guess I may need to clearly specify methods I want to listen to using `http_listenersupport(const http::method &, const std::function<void(http_request)> &)`. And it works.

So I came up with curiosity and posted my question in this thread.
